sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/util/IdentityUtil.java

```diff
@@ -37,11 +47,39 @@ public static String resolveTenantId(String currentTenantId, TokenRequestContext
                     + "`adfs` tenant id and it cannot be replaced with a tenant id challenge provided via "
                     + "TokenRequestContext class. ", null));
             }
-            return CoreUtils.isNullOrEmpty(contextTenantId) ? currentTenantId
+            String resolvedTenantId =  CoreUtils.isNullOrEmpty(contextTenantId) ? currentTenantId
                 : contextTenantId;
+
+            if (resolvedTenantId != currentTenantId && !options.getAdditionallyAllowedTenants().contains(ALL_TENANTS)
+                && !options.getAdditionallyAllowedTenants().contains(resolvedTenantId)) {
```

Comment:
We should only throw if `currentTenantId` was specified.
```suggestion
            if (currentTenantId != null && resolvedTenantId != currentTenantId && !options.getAdditionallyAllowedTenants().contains(ALL_TENANTS)
                && !options.getAdditionallyAllowedTenants().contains(resolvedTenantId)) {
```

Reply:
Also, tenant id comparison should be case insensitive. I confirmed with the AAD team that both the domain form and GUID form of tenant IDs are case insensitive.

Reply:
Yeah, the parent if already validated that currentTenantId != null.
Also, updated the underlying DS to ensure case insensitive comparison happens and also added tests for it.